pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -153,95 +153,97 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, status.APIServerError(err, "failed to get RepoSync")
 	}
 
+	currentRS := rs.DeepCopy()
+	var newValuesFileRefs []v1beta1.ValuesFileRef
+
 	if !rs.DeletionTimestamp.IsZero() {
 		// Deletion requested.
 		// Cleanup is handled above, after the RepoSync is NotFound.
 		r.logger(ctx).V(3).Info("Sync deletion timestamp detected")
-	}
-
-	currentRS := rs.DeepCopy()
-
-	if err := r.watchConfigMaps(rs); err != nil {
-		r.logger(ctx).Error(err, "Error watching ConfigMaps")
-		reposync.SetStalled(rs, "ConfigMapWatch", err)
-		_, updateErr := r.updateStatus(ctx, currentRS, rs)
-		if updateErr != nil {
-			r.logger(ctx).Error(updateErr, "Error setting status")
+	} else {
```

Comment:
Long else statements are a code smell. Can you extract the else stuff into another method or two?

I'm thinking maybe validateNamespaceName, IsDNS1123Subdomain, and validateSpec can all just go under validateSpec. And then copyConfigMapsToCms should probably move into upsertManagedObjects.

Reply:
The gotcha here is `r.setDefaultHelmDataKey(rs)`, which seems like a smell. We shouldn't be modifying the rs here. That's what got us into the trouble needing the currentRS. If we avoid modifying the rs here, we can punt the DeepCopy into the upsert method later. The default helm key shouldn't be applied back to the rs anway. It should just be an internal default we use when it's not specified.

Reply:
I'll make a PR to extract the setDefaultHelmDataKey. It's a little involved.

Reply:
Thanks. I will extract other validations into a separate method and then we can coordinate with the conflict.

Reply:
Refactored to make it less lengthy. As we discussed offline, it's not easy to avoid setting default helm data key in rs, so I ended up moving it above and reorganize the validation and upsert functions.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/803

Reply:
rebased